python/paddle/fluid/framework.py

```diff
@@ -2759,6 +2759,10 @@ def __init__(self):
 
         # use Deep gradient comrepssion or not
         self._enable_dgc = False
+        self._nccl_comm_num = 1
+        self._use_hierarchical_allreduce = False
+        self._hierarchical_allreduce_inter_nranks = 0
+        self._hierarchical_allreduce_exter_nranks = 0
```

Comment:
Why add those attributions here? Those attributions are not related program.

Reply:
It's needed in transpiler and we want the use set these attributes once.

Reply:
You can see 301~310 lines.

Reply:
I think it is not appropriate that add those options in Program. It should be fixed in the next PR.

Reply:
https://github.com/PaddlePaddle/Paddle/issues/17617